### PR TITLE
Execute unit tests several times in to prevent flakes

### PR DESCRIPTION
**Pull request by c0d1ngm0nk3y**
## Summary

To prevent flakes in the unit tests, we can just run them 5 times in PR validation. The run time should't be a issue and this would make it much harder for flakes to sneak in.

## Use Cases

In `go` flakes in the tests can happen quite easy since looping over a dict has not a stable order.

Motivation: https://github.com/paketo-buildpacks/npm-install/pull/689

## Checklist

* [X] I have viewed, signed, and submitted the Contributor License Agreement.
* [ ] I have linked issue(s) that this PR should close using keywords or the Github UI (See [docs](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue))
* [ ] I have added an integration test, if necessary.
* [X] I have reviewed the [styleguide](https://github.com/paketo-buildpacks/community/blob/main/STYLEGUIDE.md) for guidance on my code quality.
* [X] I'm happy with the commit history on this PR (I have rebased/squashed as needed).

### CODE REVIEW

**Comment by sophiewigmore** [comment]
🤔 I'm not against this, but I'd much rather just fix the flaky test to be deterministic than add a retry.

**Comment by c0d1ngm0nk3y** [comment]
> 🤔 I'm not against this, but I'd much rather just fix the flaky test to be deterministic than add a retry.

Oh no, this is not adding a retry. That would be giving up :) It executes the tests several times in different order and they have to be successful all the time. Imho, the only reason against such a check could be the runtime which shouldn't be a problem when we talk about unit tests for a pull requests.

That should make it much harder for flakes to come into the code base since they can be really annoying.

**Comment by c0d1ngm0nk3y** [comment]
@sophiewigmore Do you see any downside?

**Comment by robdimsdale** [comment]
> So is this something that you have seen happening or is it preventative?

There are flakes somewhere in the nodejs tests (I forget exactly which buildpack).

Personally, I feel ambivalent about this:

One the one hand, I'd love to make it harder for flakes to creep in. As someone who has had to re-trigger nodejs builds that are failing for flaky unit tests, i wish that never crept into the codebase to start with.

On the other hand, if we make this change and there are already flakes, it will essentially make it impossible to get a green build until we fix the flakes. Which we should do, of course, but we may not have the bandwidth to do that in the moment. Imagine there's a critical CVE in nodejs today - we don't have the bandwidth to dig into nodejs failures right now (we're a little short on maintainers) so we would probably end up changing the test count param back to `1` and manually retrying until it succeeded. Or worse - we would delete the flaky test rather than fixing it to ensure that we can ship the fix in a timely manner.

There's also the secondary cost that running the tests multiple times takes N times longer. But the buildpack unit tests are fairly quick so I don't see this as an issue.

On balance, I think this is a change that I support, though I probably wouldn't have made it myself.

**Review by robdimsdale — APPROVED** [review]

File: implementation/scripts/unit.sh
Comment (robdimsdale) [inline implementation/scripts/unit.sh:9]: FWIW this can be slightly simplified to: `"${GO_TEST_PARAMS:-}"

**Comment by c0d1ngm0nk3y** [comment]
> On the other hand, if we make this change and there are already flakes, it will essentially make it impossible to get a green build until we fix the flakes.

Then the fix gets out of the sudden priority which is a good think imho :)


> There's also the secondary cost that running the tests multiple times takes N times longer. But the buildpack unit tests are fairly quick so I don't see this as an issue.

I don't see a problem either and if unit tests are so slow that you cannot execute them several times, something should be changed anyway.

**Comment by c0d1ngm0nk3y** [comment]
> @c0d1ngm0nk3y So is this something that you have seen happening or is it preventative?

In `npm-install` there was a flake for months and it seemed that nobody bothered to much. It was a bit annoying to fix it. So I thought this might be a cheap way to prevent flakes and also give some priority to fixing them.

**Comment by c0d1ngm0nk3y** [comment]
If no one has objections, how to get this merged?

**Review by robdimsdale — APPROVED** [review]

**Comment by robdimsdale** [comment]
@c0d1ngm0nk3y I've approved it and yet it to auto merge. But I can't rebase it for you. Once you rebase this PR it should be auto-merged.

**Pull request merged**
